### PR TITLE
Fix useResizeObserver loop limit exceeded warning

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/1924

Current:
Use a raf to avoid unobserving and reobserving constantly

Old:
Uses solution from https://blog.elantha.com/resizeobserver-loop-limit-exceeded/
Story added is based on the code from the blog and it reproduces the issue. You can see this for yourself once this PR builds.

Someone pinged on the issue again and our project's complain about this periodically. I had a solution laying around so put it into a PR. This isn't a priority for our next release, but is a general nice to have.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Make sure ActionGroups still collapse as expected. Shouldn't see any warnings about loop limit exceeded.
Other components using it that should be verified in testing:
ActionGroup
ScrollView
SearchAutocomplete
Breadcrumbs
ButtonGroup
CardBase
ColorWheel
ComboBox
DatePicker
Picker
SearchWithin
Tabs

### CODE REVIEW

File: packages/@react-aria/utils/src/useResizeObserver.ts [hunk 1]
```diff
@@ -24,14 +24,20 @@ export function useResizeObserver<T extends HTMLElement>(options: useResizeObser
...
           return;
         }
 
         onResize();
+
+        resizeObserverInstance.unobserve(element);
+
+        requestAnimationFrame(() => {
+          resizeObserverInstance.observe(element);
+        });
```
Comment (devongovett) [inline packages/@react-aria/utils/src/useResizeObserver.ts:38]: 😬 

this seems... bad?

Comment (snowystinger) [reply packages/@react-aria/utils/src/useResizeObserver.ts:38]: Yeah, but apparently this is the way.
The link in the description does a better job of explaining than I can do, especially some 12 months after I last looked into it. The crux of it is
> The solution is to suspend observation and restart on the next animation frame, but only if the element resized while the handler was executing.

Which now that I'm reading that, looks like I missed an if guard checking the original size and new size. I'll update shortly, it's not necessary for today's testing as it is only a warning and there haven't been a ton of requests to deal with it.

Comment (devongovett) [reply packages/@react-aria/utils/src/useResizeObserver.ts:38]: still, feels like if we are hitting an error here we are probably doing something else wrong? like, this seems like a bandaid?

Comment (snowystinger) [reply packages/@react-aria/utils/src/useResizeObserver.ts:38]: Honestly, it appears to be a browser issue if you read the rest of that post I put in the description.

**Comment by delphineli** [comment]
Hi anyone knows when this pr will be merge please ?

**Review by LFDanLu — APPROVED** [review]
LGTM, confirmed the loop limit warning appeared before the fix in the new resize observer stories and goes away with the new changes. Also verified that other components seem to collapse/resize just fine still

**Review by ktabors — APPROVED** [review]
The code makes sense and the new test prevents the error in Chrome, Firefox, and Safari

**Comment by evargast** [comment]
@snowystinger this is great, thank you !

**Pull request merged**